File: 1201-1300/1282-Group the People Given the Group Size They Belong To/1282-Group the People Given the Group Size They Belong To.cpp
class Solution {
public:
    vector<vector<int>> groupThePeople(vector<int>& groupSizes) {
        unordered_map<int, vector<int>> table;
        vector<vector<int>> result;
        for (int i = 0; i < groupSizes.size(); ++i) {
            if (groupSizes[i] == 1) {
                result.push_back({i});
            }
            else {
                table[groupSizes[i]].push_back(i);
            }
        }
        
        for (auto& p : table) {
            for (int i = 0; i < p.second.size(); i += p.first) {
                result.push_back(vector<int>(p.second.begin() + i, p.second.begin() + i + p.first));
            }
        }
        return result;
    }
};

File: 1201-1300/1282-Group the People Given the Group Size They Belong To/1282-Group the People Given the Group Size They Belong To.py
import collections
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        table = collections.defaultdict(list)
        result = []
        for i, gs in enumerate(groupSizes):
            if gs == 1:
                result.append([i])
            else:
                table[gs].append(i)
                
        for k, v in table.items():
            for i in range(0, len(v), k):
                result.append(v[i: i + k])
        return result
